hyperspy/signal.py

```diff
@@ -4317,7 +4323,11 @@ def integrate1D(self, axis, out=None):
         if is_binned(self, axis=axis):
         # in v2 replace by
         # self.axes_manager[axis].is_binned
-            return self.sum(axis=axis, out=out)
+            with warnings.catch_warnings():
```

Comment:
Isn't this catch obsolete as this warning should be raised only if `is_binned = False`: https://github.com/hyperspy/hyperspy/blob/914bebd2944f3d63d9ffe02a52aa11d0f311e5a9/hyperspy/signal.py#L3700

Reply:
Only open point remaining.

Reply:
Thanks, your are both correct and this is now fixed in 2e757e9.